### ISSUE TITLE
Deprecate twisted.mail.protocols.SSLContextFactory

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4963]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4963|
|Type|task|
|Created|2011-03-18 04:48:12Z|
|Branch|https://github.com/twisted/twisted/tree/deprecate-redundant-ssl-context-4963|

`SSLContextFactory` provides no particular value over something like `twisted.internet.ssl.DefaultOpenSSLContextFactory` (in fact, less, since it allows SSLv2).  We should change our internal usage of the former (probably to the latter) and deprecate the former.

Attachments:

* [4963.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/724/724b097401e5ac51537dd27a2538a79f564d2f37/61c237a8015ef27debf1d6a28cfb314f7ddb171a.patch) (2327 bytes) - added by moijes12 on 2012-05-01 08:06:14Z - 
* [doubt.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/724/724b097401e5ac51537dd27a2538a79f564d2f37/3a9af30baf109d054bde1626528cae7746901d3b.patch) (3068 bytes) - added by moijes12 on 2012-05-03 15:21:42Z - 
* [4963.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/724/724b097401e5ac51537dd27a2538a79f564d2f37/582e8eab48dd5d01a49beb125a6d387f92cf95be.patch) (6321 bytes) - added by moijes12 on 2012-07-05 03:12:21Z - The tests for pop3sBackwardCompatibilty now use DefaultOpenSSLContextFactory. For this, a copy of server.pem has been placed in twisted/mail/test. All other changes that were supplied in the first patch are in here too. So, now, SSLContextFactory has been deprecated and its usage has been replaced with DefaultOpenSSLContextFactory.

<details><summary>Searchable metadata</summary>

```
trac-id__4963 4963
type__task task
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_deprecate_redundant_ssl_context_4963 branches/deprecate-redundant-ssl-context-4963
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__easy easy
time__1300423692000000 1300423692000000
changetime__1342297279000000 1342297279000000
version__None None
owner__exarkun exarkun
cc__moijes12
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @moijes12|
|-|-|

Thanks for working on this, moijes12.  It looks like your patch does a good job addressing the second half of this ticket, but I don't think it covers the first half.

Twisted itself should never trigger its own deprecation warnings.  Therefore, when deprecating a Twisted API, we also need to remove all usages of that API within Twisted itself.  (You may have noticed that we don't always do a good job of this, by the number of warnings emitted when running the test suite.  This is an area where we need to improve.)

Apart from that:

  1. make sure you wrap lines at 80 columns or before
  1. name tests like `test_foo` instead of `testFoo`
  1. Don't start test method docstrings with "Test ..."

Thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="moijes12's avatar" src="https://avatars.githubusercontent.com/u/4566851?s=50" width="50" height="50">](https://github.com/moijes12)|@moijes12 removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

(In [34763]) Branching to 'deprecate-redundant-ssl-context-4963'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

(In [34765]) Just use the server.pem from the source tree, no need to copy it to a temporary file first.

refs [#4963](https://github.com/twisted/twisted/issues/4963)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

(In [34766]) - Rewrite the test method docstring in the conventional style
- Remove the call of SSLContextFactory, attribute access is enough
- wrap the long warning assertion line to < 80 columns

refs [#4963](https://github.com/twisted/twisted/issues/4963)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

(In [34767]) - Remove trailing whitespace
- Wrap long deprecation line to < 80 columns

refs [#4963](https://github.com/twisted/twisted/issues/4963)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set owner to @exarkun|
|-|-|

Thanks!  I made a few small changes to the patch, mostly whitespace.  Otherwise it looks good, so I'll apply it.  Thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun set status to `closed`|
|-|-|

(In [34769]) Merge deprecate-redundant-ssl-context-4963

Author: moijes12
Reviewer: exarkun
Fixes: [#4963](https://github.com/twisted/twisted/issues/4963)

Deprecate `twisted.mail.protocols.SSLContextFactory`.
